presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1003,6 +1021,22 @@ public void dropTable(HiveIdentity identity, String databaseName, String tableNa
         }
     }
 
+    private static void deleteDir(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path, boolean recursive)
+    {
+        try {
+            hdfsEnvironment.getFileSystem(context, path).delete(path, recursive);
+        }
+        catch (Exception e) {
```

Comment:
Nit: this should be `IOException | RuntimeException`. We avoid catching `Exception` since that can hide unexpected checked exceptions like `InterruptedException` (if the code inside was changed to throw that).